// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package feedwire;

import "components/feed/core/proto/v2/wire/capability.proto";
import "components/feed/core/proto/v2/wire/client_info.proto";
import "components/feed/core/proto/v2/wire/client_user_profiles.proto";
import "components/feed/core/proto/v2/wire/consistency_token.proto";
import "components/feed/core/proto/v2/wire/diagnostic_info.proto";
import "components/feed/core/proto/v2/wire/feed_query.proto";

option optimize_for = LITE_RUNTIME;

message FeedRequest {
  optional ClientInfo client_info = 1;
  optional FeedQuery feed_query = 2;
  repeated Capability client_capability = 4;
  optional ConsistencyToken consistency_token = 5;
  optional DiagnosticInfo diagnostic_info = 14;
  optional ClientUserProfiles client_user_profiles = 16;
}
